kbmd/plugin.c

```diff
@@ -0,0 +1,1240 @@
+/*
```

Comment:
The whole plugin interface is clunky imo (calling out to a separate executable makes things a lot more complex that is needed in a number of ways), but the design was effectively imposed on me (and the freedom to redo it in a less clunky manner happened at too late a stage), so the focus was on trying to keep things as simple as possible.  Judging by the sheer number of articles I've seen over the years on how no one can seem to correctly parse JSON, I wanted to avoid having to do that as much as possible (or at least confine it to node.js to node.js on both ends via the json command -- that way at least, any parsing bugs would be largely consistent on both sides).

Reply:
fair enough.